pages/DSNP/Identity.md

```diff
@@ -24,3 +24,17 @@ the ability to delegate permission to others to announce content on the user's b
 * An owner MUST be able to delegate permission to announce on their behalf to other parties.
 * A user MUST be able to revoke delegated permissions.
 * Announcements from a delegate MUST be able to be verified as to which delegate made the specific announcement.
+* Delegation revocation MUST NOT be retroactive.
+
+
+## Non-normative
+
+### Retroactive Revocation of Delegation
+
+There are many times when someone would desire retroactive revocation of delegation.
+If a key was found to have been compromised at an earlier time for example.
+However, retroactive revocation is much more difficult from a caching and performance perspective.
+Instead (reverting any undesirable Announcements)(Announcements.md#reverting-an-announcement) such as through [Tombstones](Types/Tombstone.md) allows a user to choose the specific events that need reverting and notify the network of that change.
+
+To this end, any [Batch Publications](BatchPublications.md) can always be validated from the perspective of the time they were published instead of at read time.
+The validity of a Batch is thus immutable.
```

Comment:
I like this approach how will require some discussion around how to track such edge cases to know delegation was removed after the published time but not important for this PR

Reply:
Tried adding some steps below to make it more clear.